Initial setup of Javalin

ndoschek opened the pull request:
- Load model at request
- Use EMFJsonConverter
- First get request implemented: get model via /api/model/filepath (e.g. localhost:8081/api/model/Coffe.ecore)

ndoschek committed:
Initial setup of Javalin
- Load model at request
- Use EMFJsonConverter
- First get request implemented: get model via /api/model/filepath

tortmayr commented on com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerModule.java:
```diff
@@ -0,0 +1,40 @@
+
+import io.javalin.Javalin;
+
+public class ModelServerModule extends AbstractModule {
```
Maybe we should combine the `ModelServerModule` and the `EMFModelServerModule`?
Or at least make the `EMFModelServerModule` a Subclass of `ModelServerModule`?

ndoschek replied:
Yes, sounds good! Will work that in!

tortmayr commented on com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerEntrypoint.java:
```diff
@@ -0,0 +1,58 @@
+public class ModelServerEntrypoint implements AppEntrypoint {
+	
+	private Javalin app;
+	private static final Logger logger = LoggerFactory.getLogger(ModelServerLauncher.class);
```
Maybe we should agree on common naming for the Logger instances.
I used LOG but I don't have a preference, I'd just like it to be consistent.

edgarmueller commented on com.eclipsesource.modelserver.common/src/main/java/com/eclipsesource/modelserver/common/AppEntrypoint.java:
```diff
@@ -0,0 +1,20 @@
+ *******************************************************************************/
+package com.eclipsesource.modelserver.common;
+
+public interface AppEntrypoint {
```
Minor: casing, should read `AppEntryPoint` (here as well as anywhere else).

edgarmueller commented on com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerController.java:
```diff
@@ -0,0 +1,83 @@
+import io.javalin.Context;
+import io.javalin.apibuilder.CrudHandler;
+
+public class ModelServerController implements CrudHandler {
```
I'd rename this to `ModelController`

edgarmueller commented on com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerEntrypoint.java:
```diff
@@ -0,0 +1,58 @@
+	private Javalin app;
+	private static final Logger logger = LoggerFactory.getLogger(ModelServerLauncher.class);
+
+	@SuppressWarnings("rawtypes")
```
No need for `rawtypes` here

edgarmueller commented on com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerStartup.java:
```diff
@@ -0,0 +1,35 @@
+		Optional<AppEntrypoint> entryPoint = Optional.ofNullable(entrypoints.get(entrypointType));
+		entryPoint.orElseThrow(() -> new RuntimeException("Entrypoint not defined")).boot(args);;
+	}
+}
```
Newline missing

edgarmueller commented on com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerStartup.java:
```diff
@@ -0,0 +1,35 @@
+
+	public void boot(EntrypointType entrypointType, String[] args) {
+		Optional<AppEntrypoint> entryPoint = Optional.ofNullable(entrypoints.get(entrypointType));
+		entryPoint.orElseThrow(() -> new RuntimeException("Entrypoint not defined")).boot(args);;
```
Unnecessary semicolon

tortmayr reviewed (commented):
Looks good to me. Only minors/suggestions.

I think the mock ModelServer interface and EMFModelServer class can be removed now. Maybe the mechanism for caching already loaded models could be migrated to the ModelServerController

ndoschek committed:
- Configure JavalinJackson for emfjson
- Remove ModelServer interface and EMFModelServer class
- Combine ModelServerModule and EMFModelServerModule
- Remove ExampleModelServerModule, ad EPackageConfigurations instead to ModelServerLauncher
- Rename logging instance to LOG

edgarmueller reviewed (approved):
Great work, thanks! I've pointed out only minor things, I'll probably fix them myself and push a PR.
